dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java

```diff
@@ -367,8 +360,24 @@ private Container geContainerById(final String containerIdOrPath, final User use
     }
 
     private boolean needParseContainerPrefix(final Container container, final String uniqueId) {
-        final String containerIdOrPath = (container instanceof FileAssetContainer) ?
-                ((FileAssetContainer) container).getPath() : container.getIdentifier();
+        String containerIdOrPath = null;
+
+        if (FileAssetContainerUtil.getInstance().isFileAssetContainer(container)) {
+            final Host host;
+            try {
+                host = APILocator.getHostAPI().findParentHost(container, APILocator.systemUser(), false);
+
+                containerIdOrPath = this.site.getIdentifier().equals(host.getIdentifier()) ?
+                        ((FileAssetContainer) container).getPath() :
+                        FileAssetContainerUtil.getInstance().getFullPath((FileAssetContainer) container);
+            } catch (DotDataException | DotSecurityException e) {
```

Comment:
Is this an outside case or are we using exceptions as control statements?

Reply:
this was remove https://github.com/dotCMS/core/pull/18033/commits/286b9c09f52d6e0bb78ec9761ec3000c06d7f38a#diff-c7538880be24298c9c76f10f1c8efaffL373